Advanced/Play Asset Delivery/Assets/PlayAssetDelivery/Editor/BuildScriptPlayAssetDelivery.cs

```diff
@@ -0,0 +1,172 @@
+    /// <summary>
+    /// In addition to the Default Build Script behavior (building AssetBundles), this script creates prepares bundled content for custom asset pack creation
+    /// https://docs.unity3d.com/Manual/play-asset-delivery.html#custom-asset-packs. 
+    /// 
+    /// At build time the AddressablesAssetPostProcessor will temporarily move all built content to 'Assets/StreamingAssets'.
+    /// This means that any Addressables content will be automatically be included in the streaming assets pack even if they are not assigned to a custom asset pack. 
+    /// 
+    /// We also assign any content marked for install-time delivery to the streaming assets pack. In most cases the streaming assets pack will use "install-time" delivery, 
+    /// but in large projects it may use "fast-follow" delivery instead. For more information see https://docs.unity3d.com/Manual/play-asset-delivery.html#generated-asset-packs.
+    ///  
+    /// For other delivery types bundles will be moved into a {asset pack name}.androidpack folder located in 'Assets/PlayAssetDelivery/CustomAssetPackContent'.
+    /// A 'build.gradle' file will also be created in each .androidpack folder unless the delivery type is set to "none". 
+    /// In this case Unity automatically assumes that any asset packs without a 'build.gradle' file use "on-demand" delivery.
+    /// </summary>
+    [CreateAssetMenu(fileName = "BuildScriptPlayAssetDelivery.asset", menuName = "Addressables/Custom Build/Play Asset Delivery")]
+    public class BuildScriptPlayAssetDelivery : BuildScriptPackedMode
+    {
+        /// <inheritdoc/>
+        public override string Name
+        {
+            get { return "Play Asset Delivery"; }
+        }
+
+        static string m_AssetPackFolderName = "CustomAssetPackContent";
+        static string m_RootDirectory = "Assets/PlayAssetDelivery";
+        static string PackContentRootDirectory
+        {
+            get { return $"{m_RootDirectory}/{m_AssetPackFolderName}"; }
+        }
+
+        protected override TResult DoBuild<TResult>(AddressablesDataBuilderInput builderInput, AddressableAssetsBuildContext aaContext)
+        {
+            TResult result = base.DoBuild<TResult>(builderInput, aaContext);
+            AddressableAssetSettings settings = aaContext.Settings;
+            var assetPackToDataEntry = new Dictionary<string, CustomAssetPackDataEntry>();
+
+            // Reset schema data if the settings file was deleted
+            bool resetSchemaData = !CustomAssetPackSettings.SettingsExists ? true : false;
+
+            List<CustomAssetPackEditorInfo> customAssetPacks = CustomAssetPackSettings.GetSettings().CustomAssetPacks;
+            
+            // Clear out the 'Assets/PlayAssetDelivery/CustomAssetPackContent' directory
+            if(Directory.Exists(PackContentRootDirectory))
+                AssetDatabase.DeleteAsset(PackContentRootDirectory);
+            AssetDatabase.CreateFolder(m_RootDirectory, m_AssetPackFolderName);
```

Comment:
Looks like a typo, it should be: AssetDatabase.CreateFolder(PackContentRootDirectory, m_AssetPackFolderName);

Reply:
Hmm, are you sure it was a typo? PackContentRootDirectory is $"{m_RootDirectory}/{m_AssetPackFolderName}", so now this "CreateFolder" call will try to create $"{m_RootDirectory}/{m_AssetPackFolderName}/{m_AssetPackFolderName}" directory, which is probably not correct. Also the original concern was that looking at AssetDatabase.CreateFolder it seems that if the directory specified in the first parameter does not exist, the directory name specified in the second parameter is not created either. So after your change, even if duplicating m_AssetPackFolderName in the path is what we want, this folder will not be created, since at this point PackContentRootDirectory will always be empty (since it is deleted in a line above). This is the concern I had in my original comment.

Reply:
Oh nevermind, yes you are right. The m_RootDirectory (Assets/PlayAssetDelivery) will always exist in the project. It also contains all the script files. Deleting and creating the PackContentRootDirectory(Assets/PlayAssetDelivery/CustomAssetPackContent) for each new build is the behavior we want. It will contain all the asset pack folders and bundles assigned to the custom asset packs.